### DISCUSSION TITLE
Changing contributors license agreement for ANTLR going forward

### DISCUSSION THREAD

**Discussion by parrt** [antlr/antlr4#3589] [General]
I propose that we simplify and standardize the [contributors license agreement for ANTLR](https://github.com/antlr/antlr4/blob/master/contributors.txt) (*CLA*) from the custom thing we've been using to that used by the Linux Foundation [Developer Certificate of Origin v1.1](https://developercertificate.org/).  Over the years, it has discouraged some contributions; some individuals and various company employees have been unwilling or unable to sign the agreement because of the indemnification cause.

Does anybody see any bears on the woods? Any issues with changing to the Linux CLA? @sharwell @ericvergnaud @mike-lischke @jcking @KvanTTT

**Comment by mike-lischke** [comment]
Fine by me 🤜

**Reply by KvanTTT** [reply]
For me as well.

**Reply by jcking** [reply]
Sounds good to me as well.

**Reply by ericvergnaud** [reply]
Fine by me, but I do see a potential issue where contributors have signed _another_ licensing agreement, so to be fully protected, we would need all contributors to agree...

**Reply by parrt** [reply]
Yeah,I was wondering about that.  probably not practical to get all of those agreements from the last decade as people have disappeared and so on.  Given that the new CLA is in some sense less strict on them, I don't think they would care. My plan would be to rename the current contributors.txt file so that those that agreed under that CLA would still have the same agreement. Only new contributors would add to the new contributors.txt file. Do you think that's reasonable?

**Reply by mike-lischke** [reply]
You could apply a time tag (say a year) where people can open disputes, if they don't agree with the new CLA. After that you can remove the old one (to avoid keeping it forever).

**Reply by parrt** [reply]
Well, I'd rather not invite disputes or anything that muddies the license so I will probably just keep the old file around and we can start with the new Linux distribution CLA.  still waiting to hear from @sharwell so I'll hold off for now.

**Comment by parrt** [comment]
Had a conversation with Sam and he has no objections. He did suggest that we look at automating this process and apparently there is a way to do that at github now.  Probably the simplest answer is just to notify everyone in the PR template that they are subject to this linux certificate of origin. I will then move the previous notifications file to one that indicates it's historical. We can do this coincidental with the 4.10 release, hence, I will make a PR in dev.  

https://github.com/antlr/antlr4/pull/3598

**Reply by parrt** [reply]
Ok, it looks like I'm going to require the `-s` option to enforce the signature of each commit.  I just have to turn on the github option for the repo.  See

https://wiki.linuxfoundation.org/dco

and

https://github.com/apps/dco

**Reply by parrt** [reply]
I have pulled this into the dev branch and made a note in the release documentation to ensure that the GitHub DCO enforcement is turned on when we release.
